### DISCUSSION TITLE
Search: pre-indexing directories

### DISCUSSION THREAD

**Discussion by Topener** [docsifyjs/docsify#1596] [Q&A] [answered]
Hi,

I've implemented the search plugin on the documentation page, it works finally (Thanks!), but now I noticed it doesn't index other sidebars before I visit them. 

In the case of my page (https://sharp-lovelace-ddcd32.netlify.app/) I have a page under `/module-development/'`, once I've visited this page and go back to home I can search within the content of that page, but not before.

Is there a way to pre-populate the search so it can immediately search in those directories?

**Comment by Topener** [comment]
For now I fixed it by just linking the sub-documents in the root sidebar, but want to know if there are options.

**Comment by sy-records** [comment, marked as answer]
You are doing the right thing.

The index is generated for searching when first visited, instead of having to visit the page to search

![image](https://user-images.githubusercontent.com/33931153/122847238-f27ff200-d339-11eb-9c95-aa416af85e7a.png)
